### PR TITLE
[Imp] improve product form design

### PR DESCRIPTION
**Pull request by atp-odoo**
Task: https://www.odoo.com/web#id=13764&view_type=form&model=project.task&action=327
Pad: https://pad.odoo.com/p/r.XGBE0Yi5fflqJN61

### CODE REVIEW

File: addons/event_sale/views/product.xml [hunk 1]
```diff
@@ -5,10 +5,10 @@
              <field name="model">product.template</field>
              <field name="inherit_id" ref="product.product_template_form_view" />
              <field name="arch" type="xml">
-                <div name="options" position="inside">
+                <div name="event_ok" position="replace">
                     <div>
```
Comment (atp-odoo) [inline addons/event_sale/views/product.xml]: find other alternative ,. may be use priority of view.

File: addons/product/product_view.xml [hunk 2]
```diff
@@ -63,63 +63,58 @@
                                     <field name="sale_ok"/>
                                     <label for="sale_ok"/>
                                 </div>
+                                <div name="purchase_ok" invisible="1"/>
+                                <div name="event_ok" invisible="1"/>
+                                <div name="hr_expense_ok" invisible="1"/>
                             </div>
```
Comment (atp-odoo) [inline addons/product/product_view.xml]: use inherit view with priority instead of this change.

File: addons/stock/product.py [hunk 3]
```diff
@@ -201,6 +201,12 @@ def _product_available_text(self, cr, uid, ids, field_names=None, arg=False, con
             res[product.id] = str(product.qty_available) +  _(" On Hand")
         return res
 
+    def _reordering_rules_count(self, cr, uid, ids, field_name, arg, context=None):
+        res = dict([(id, 0) for id in ids])
+        for product in filter(lambda x: x.orderpoint_ids, self.browse(cr, uid, ids, context=context)):
+            res[product.id] = len([p.product_id.id for p in product.orderpoint_ids])
+        return res
```
Comment (atp-odoo) [inline addons/stock/product.py]: do it simpler. directly we can find length len(product.orderpoint_ids)

File: addons/stock/product.py [hunk 4]
```diff
@@ -259,8 +265,22 @@ def _product_available_text(self, cr, uid, ids, field_names=None, arg=False, con
         'location_id': fields.dummy(string='Location', relation='stock.location', type='many2one'),
         'warehouse_id': fields.dummy(string='Warehouse', relation='stock.warehouse', type='many2one'),
         'orderpoint_ids': fields.one2many('stock.warehouse.orderpoint', 'product_id', 'Minimum Stock Rules'),
+        'reordering_count': fields.function(_reordering_rules_count, type='integer', string="Reordering Rules"),
     }
 
+    def action_virtual_stock(self, cr, uid, ids, context=None):
+        view_id = self.pool['ir.model.data'].xmlid_to_res_id(cr, uid, 'stock.product_product_virtual_stock', raise_if_not_found=True)
```
Comment (atp-odoo) [inline addons/stock/product.py]: give proper name. product_view_id

File: addons/stock/product.py [hunk 5]
```diff
@@ -365,7 +385,12 @@ def _product_available_text(self, cr, uid, ids, field_names=None, arg=False, con
             res[product.id] = str(product.qty_available) +  _(" On Hand")
         return res
 
-
+    def _reordering_rules_count(self, cr, uid, ids, field_name, arg, context=None):
+        res = dict([(id, 0) for id in ids])
+        product_tmpl_id = self.browse(cr, uid, ids, context=context)
```
Comment (atp-odoo) [inline addons/stock/product.py]: product_template instead of tmpl_id . browse not return id.

File: addons/stock/product_view.xml [hunk 6]
```diff
@@ -217,6 +209,32 @@
             </field>
         </record>
 
+        <record model="ir.ui.view" id="product_template_virtual_stock">
+            <field name="name">product.template.virtual.tree</field>
```
Comment (atp-odoo) [inline addons/stock/product_view.xml]: follow guidlines for view id name. id name like : view_product_template_tree..

File: addons/event_sale/views/product.xml (hunk 1 above)
Comment (ppa-odoo) [inline addons/event_sale/views/product.xml]: @atp-odoo thankyou for suggestion,
done.

File: addons/product/product_view.xml (hunk 2 above)
Comment (ppa-odoo) [inline addons/product/product_view.xml]: @atp-odoo removed invisible div tag.

File: addons/stock/product.py (hunk 3 above)
Comment (ppa-odoo) [inline addons/stock/product.py]: done.

File: addons/stock/product.py (hunk 5 above)
Comment (ppa-odoo) [inline addons/stock/product.py]: done.

File: addons/stock/product.py (hunk 4 above)
Comment (ppa-odoo) [inline addons/stock/product.py]: changed view_id name.

File: addons/stock/product_view.xml (hunk 6 above)
Comment (ppa-odoo) [inline addons/stock/product_view.xml]: i changed view id name as per ur suggestions.

File: addons/account/views/product_view.xml [hunk 7]
```diff
@@ -9,16 +9,18 @@
...
-                            <div><field name="categ_id" colspan="3" nolabel="1"/></div>
+                        <group name="product_category">
+                            <group name="category">
+                                <label for="categ_id" string="Internal Category"/>
+                                <div><field name="categ_id" colspan="3" nolabel="1"/></div>
+                            </group>
                         </group>
                         <group name="properties">
-                            <group>
+                            <group name="sale_tax_part">
```
Comment (atp-odoo) [inline addons/account/views/product_view.xml]: why did you give group name  which are not used ?

File: addons/account/views/product_view.xml [hunk 8]
```diff
@@ -9,16 +9,18 @@
...
                                 <field name="property_account_income_id" domain="[('internal_type','=','other'),('deprecated','=',False)]" groups="account.group_account_user"/>
                                 <field name="taxes_id" colspan="2" widget="many2many_tags"/>
                             </group>
-                            <group>
+                            <group name="purchase_tax_part">
```
Comment (atp-odoo) [inline addons/account/views/product_view.xml]: same here .no need to use group name.

File: addons/account_asset/views/account_asset_view.xml [hunk 9]
```diff
@@ -255,12 +255,12 @@
...
-                    <field name="asset_category_id" domain="[('type','=','purchase')]" context="{'default_type':'purchase'}"/>
-                    <field name="deferred_revenue_category_id" domain="[('type','=','sale')]" context="{'default_type':'sale'}"/>
-                </group>
-            </xpath>
+            <field name="property_account_expense_id" position="before">
+                <field name="asset_category_id" domain="[('type','=','purchase')]" context="{'default_type':'purchase'}"/>
+            </field>
+            <field name="property_account_income_id" position="before">
+                <field name="deferred_revenue_category_id" domain="[('type','=','sale')]" context="{'default_type':'sale'}"/>
+            </field>
```
Comment (atp-odoo) [inline addons/account_asset/views/account_asset_view.xml]: follow pep8 guidlines. space after comma.

File: addons/event_sale/views/product.xml [hunk 10]
```diff
@@ -1,14 +1,15 @@
...
              <field name="model">product.template</field>
              <field name="inherit_id" ref="product.product_template_form_view" />
+             <field name="priority" eval="3"/>
              <field name="arch" type="xml">
                 <div name="options" position="inside">
                     <div>
-                        <field name="event_ok" on_change="onchange_event_ok(type, event_ok, context)"/>
-                        <label for="event_ok"/>
+                        <field name="event_ok" on_change="onchange_event_ok(type, event_ok, context)" groups="base.group_no_one"/>
+                        <label for="event_ok" groups="base.group_no_one"/>
```
Comment (atp-odoo) [inline addons/event_sale/views/product.xml]: Give groups to div tag instead of field and label.

File: addons/account/views/product_view.xml (hunk 7 above)
Comment (guewen) [inline addons/account/views/product_view.xml]: names are very useful when inheriting the views, it is really appreciated to have a name on which we can rely (rather than strings which are expected to change sometimes)

Comment (atp-odoo) [inline addons/account/views/product_view.xml]: i know names are always usefull when we inheriting views, but here no need to change that things for this task. it's useless change. here field name is enough to add something.

File: addons/product/product_view.xml [hunk 11]
```diff
@@ -66,60 +66,49 @@
...
+                                    <group>
+                                        <field digits="(14, 3)" name="weight" string="Weight" attrs="{'invisible':[('type','=','service')]}"/>
+                                        <field name="company_id" groups="base.group_multi_company" options="{'no_create': True}"/>
                                     </group>
                                 </group>
-                                <group name="packaging" string="Packaging" attrs="{'invisible':[('type','=','service')]}" groups="product.group_stock_packaging" colspan="4">
-                                    <field name="packaging_ids" string="Configurations" context="{'tree_view_ref':'product.product_packaging_tree_view_product', 'form_view_ref': 'product.product_packaging_form_view_without_product'}"/>
+                                <group name="inventory" colspan="4">
+                                    <group name="lot" invisible="1">
+                                    </group>
```
Comment (atp-odoo) [inline addons/product/product_view.xml]: no need to set lot group with invisible. you can use inventory group.

File: addons/product/product_view.xml [hunk 12]
```diff
@@ -66,60 +66,49 @@
...
+                                <separator string="Suppliers"/>
+                                <field name="seller_ids"/>
+                                <separator string="Packagings" groups="product.group_stock_packaging" attrs="{'invisible':[('type','=','service')]}"/>
+                                <field name="packaging_ids" attrs="{'invisible':[('type','=','service')]}"
+                                    groups="product.group_stock_packaging" context="{'tree_view_ref':'product.product_packaging_tree_view_product', 'form_view_ref': 'product.product_packaging_form_view_without_product'}"/>
                             </page>
                             <page string="Sales" attrs="{'invisible':[('sale_ok','=',False)]}" name="sales">
-                                <group name="sale">
-                                    <group name="sale_condition" string="Sale Conditions" colspan="3">
+                                <group name="sale" colspan="4">
```
Comment (atp-odoo) [inline addons/product/product_view.xml]: why need to set colspan 4? bydefault page is divided in 4 colspan.

File: addons/event_sale/models/event.py [hunk 13]
```diff
@@ -6,6 +6,15 @@
 from openerp.osv import fields as old_fields
 
 
+class event_type(models.Model):
```
Comment (mga-odoo) [inline addons/event_sale/models/event.py]: It should be inherited from product.template, so when you create a new event type it will create a new product with same name and event_ok fields set to True by default

remove code related to event_type_id from everywhere.

File: addons/product/product_view.xml [hunk 14]
```diff
@@ -66,60 +66,45 @@
...
-                                        <field name="weight"/>
-                                        <field name="weight_net"/>
+                                    <group>
+                                        <field name="company_id" groups="base.group_multi_company" options="{'no_create': True}"/>
                                     </group>
                                 </group>
+                                <group name="inventory" attrs="{'invisible':[('type', 'in', ['service', 'digital'])]}"/>
+                                <separator string="Suppliers"/>
+                                <field name="seller_ids"/>
                                 <group name="packaging" string="Packaging" attrs="{'invisible':[('type','=','service')]}" groups="product.group_stock_packaging" colspan="4">
```
Comment (mga-odoo) [inline addons/product/product_view.xml]: check for service or digital product to display packaging 
attrs="{'invisible':[('type','in',['service','digital'])]}"

File: addons/product/product_view.xml [hunk 15]
```diff
@@ -128,13 +113,19 @@
...
                                         <field name="uos_id"/>
                                         <field name="uos_coeff"/>
-                                        <field name="mes_type"/>
                                     </group>
                                 </group>
                                 <group name="website_and_pos" col="2">
                                 </group>
-                                <separator string="Description for Quotations"/>
-                                <field name="description_sale" placeholder="note to be displayed on quotations..."/>
+                                <group name="email_template_and_project" attrs="{'invisible':[('type', '!=', 'service')]}"/>
```
Comment (mga-odoo) [inline addons/product/product_view.xml]: explain a problem in code why you need this, i understand it is trick to solve a small problem of display a label,

File: addons/stock/product.py [hunk 16]
```diff
@@ -201,6 +201,12 @@ def _product_available_text(self, cr, uid, ids, field_names=None, arg=False, con
             res[product.id] = str(product.qty_available) +  _(" On Hand")
         return res
 
+    def _reordering_rules_count(self, cr, uid, ids, field_name, arg, context=None):
+        res = {}
+        for product in filter(lambda x: x.orderpoint_ids, self.browse(cr, uid, ids, context=context)):
```
Comment (mga-odoo) [inline addons/stock/product.py]: convert to read group

File: addons/stock/product.py [hunk 17]
```diff
@@ -365,7 +372,12 @@ def _product_available_text(self, cr, uid, ids, field_names=None, arg=False, con
             res[product.id] = str(product.qty_available) +  _(" On Hand")
         return res
 
-
+    def _reordering_rules_count(self, cr, uid, ids, field_name, arg, context=None):
```
Comment (mga-odoo) [inline addons/stock/product.py]: improve this method, use read group instead of search count

File: addons/stock/stock.py [hunk 18]
```diff
@@ -1571,7 +1571,7 @@ class stock_production_lot(osv.osv):
     _columns = {
         'name': fields.char('Serial Number', required=True, help="Unique Serial Number"),
         'ref': fields.char('Internal Reference', help="Internal reference number in case it differs from the manufacturer's serial number"),
-        'product_id': fields.many2one('product.product', 'Product', required=True, domain=[('type', '<>', 'service')]),
```
Comment (mga-odoo) [inline addons/stock/stock.py]: it should be not in ?

File: addons/stock/stock.py [hunk 19]
```diff
@@ -1571,7 +1571,7 @@ class stock_production_lot(osv.osv):
...
+        'product_id': fields.many2one('product.product', 'Product', required=True, domain=[('type', 'in', ['service', 'digital'])]),
```
Comment (mga-odoo) [inline addons/stock/stock.py]: it should be "not in" ?

File: addons/stock/stock.py [hunk 20]
```diff
@@ -1744,7 +1744,7 @@ def _set_product_qty(self, cr, uid, id, field, value, arg, context=None):
         'create_date': fields.datetime('Creation Date', readonly=True, select=True),
         'date': fields.datetime('Date', required=True, select=True, help="Move date: scheduled date until move is done, then date of actual move processing", states={'done': [('readonly', True)]}),
         'date_expected': fields.datetime('Expected Date', states={'done': [('readonly', True)]}, required=True, select=True, help="Scheduled date for the processing of this move"),
-        'product_id': fields.many2one('product.product', 'Product', required=True, select=True, domain=[('type', '<>', 'service')], states={'done': [('readonly', True)]}),
+        'product_id': fields.many2one('product.product', 'Product', required=True, select=True, domain=[('type', 'in', ['service', 'digital'])], states={'done': [('readonly', True)]}),
```
Comment (mga-odoo) [inline addons/stock/stock.py]: it should be "not in" ?

File: addons/website/static/src/xml/website.backend.xml [hunk 21]
```diff
@@ -2,8 +2,8 @@
 <templates id="template" xml:space="preserve">
 
     <div t-name="WidgetWebsiteButton" class="o_stat_info published">
-        <span class="o_stat_text oe_unpublish text-danger">Not Published</span>
-        <span class="o_stat_text oe_publish text-success">Published</span>
+        <span class="o_stat_text oe_unpublish text-danger">No Published<br/>in eCommerce</span>
```
Comment (mga-odoo) [inline addons/website/static/src/xml/website.backend.xml]: It should be "Not Published in eCommerce" ?

File: addons/website_sale_digital/product.py [hunk 22]
```diff
@@ -5,4 +5,52 @@
 class product_template(models.Model):
     _inherit = ['product.template']
 
-    digital_content = fields.Boolean('Digital Content', help="If checked, it will allow clients to download the product attachments when they have bought it.")
+    type = fields.Selection([('digital', 'Digital Content'),('product', 'Stockable Product'), ('consu', 'Consumable'), ('service', 'Service')], help="Consumable: Will not imply stock management for this product. \nStockable product: Will imply stock management for this product.\n Digital Product :If select digital , it will allow clients to download the product attachments when they have bought it.")
+    attachment_count = fields.Integer(compute='_compute_product_attachment_count', string="File")
+
+    @api.one
+    def _compute_product_attachment_count(self):
```
Comment (mga-odoo) [inline addons/website_sale_digital/product.py]: use api.multi and use read group

File: addons/website_sale_digital/product.py [hunk 23]
```diff
@@ -5,4 +5,52 @@
...
+            'context': "{'default_res_model': '%s','default_res_id': %d}" % (self._name, self.id),
+        }
+
+class product_product(models.Model):
+    _inherit = 'product.product'
+
+    attachment_count = fields.Integer(compute='_compute_product_attachment_count', string="File")
+
+    @api.one
+    def _compute_product_attachment_count(self):
```
Comment (mga-odoo) [inline addons/website_sale_digital/product.py]: use api.multi and use read group

File: addons/event_sale/models/event.py (hunk 13 above)
Comment (ppa-odoo) [inline addons/event_sale/models/event.py]: Inherited product.product and removed event_type_id and related code.

File: addons/product/i18n/product.pot [hunk 24]
```diff
@@ -1892,11 +1892,6 @@ msgid "Name of this kind of price."
 msgstr ""
 
 #. module: product
-#: field:product.template,weight_net:0
```
Comment (tivisse) [inline addons/product/i18n/product.pot]: Why ?

File: addons/mail/views/mail_message_views.xml [hunk 25]
```diff
@@ -99,18 +99,20 @@
...
-                                        <div t-if="record.file_type_icon.value != 'webimage'">
-                                            <img t-att-src="'/mail/static/src/img/mimetypes/' + record.file_type_icon.value + '.png'"></img>
-                                        </div>
-                                        <div t-if="record.file_type_icon.value == 'webimage' and !record.url.value">
-                                            <img t-att-src="kanban_image('ir.attachment', 'datas', record.id.value)"/>
-                                        </div>
-                                        <div t-if="record.file_type_icon.value and record.url.value">
-                                            <img t-att-src="record.url.value"/>
-                                        </div>
+                                        <a t-att-href="kanban_image_saveas('ir.attachment', 'datas', 'datas_fname', record.id.value)">
```
Comment (tde-banana-odoo) [inline addons/mail/views/mail_message_views.xml]: What ?

File: addons/membership/membership_view.xml [hunk 26]
```diff
@@ -80,24 +80,6 @@
             </field>
         </record>
 
-        <record model="ir.ui.view" id="product_template_form_view">
```
Comment (tde-banana-odoo) [inline addons/membership/membership_view.xml]: Can I configure the membership details elsewhere ?

File: addons/product/product.py [hunk 27]
```diff
@@ -345,9 +345,12 @@ class product_attribute_line(osv.osv):
     _columns = {
         'product_tmpl_id': fields.many2one('product.template', 'Product Template', required=True, ondelete='cascade'),
         'attribute_id': fields.many2one('product.attribute', 'Attribute', required=True, ondelete='restrict'),
-        'value_ids': fields.many2many('product.attribute.value', id1='line_id', id2='val_id', string='Product Attribute Value(s)'),
+        'value_ids': fields.many2many('product.attribute.value', id1='line_id', id2='val_id', string='Attribute Value(s)'),
     }
-
+    def name_search(self, cr, user, name='', args=None, operator='ilike', context=None, limit=100):
```
Comment (tde-banana-odoo) [inline addons/product/product.py]: I think this is not a good idea performance-wise.

File: addons/product/product.py [hunk 28]
```diff
@@ -472,14 +457,15 @@ def _set_product_template_field(self, cr, uid, product_tmpl_id, name, value, arg
...
-                                                "Expressed in the default unit of measure of the product..", groups="base.group_user", store=True),
-        'volume': fields.function(_compute_product_template_field, fnct_inv=_set_product_template_field, multi='_compute_product_template_field', type='float', string='Volume', help="The volume in m3.", store=True),
-        'weight': fields.function(_compute_product_template_field, fnct_inv=_set_product_template_field, multi='_compute_product_template_field', type='float', string='Gross Weight', digits_compute=dp.get_precision('Stock Weight'), help="The gross weight in Kg.", store=True),
-        'weight_net': fields.function(_compute_product_template_field, fnct_inv=_set_product_template_field, multi='_compute_product_template_field', type='float', string='Net Weight', digits_compute=dp.get_precision('Stock Weight'), help="The net weight in Kg.", store=True),
+        'standard_price': fields.property(type = 'float', digits_compute=dp.get_precision('Product Price'),
```
Comment (tde-banana-odoo) [inline addons/product/product.py]: Why reverting something @tivisse changed a few days ago ?

File: addons/event/data/event_demo.xml [hunk 29]
```diff
@@ -5,27 +5,6 @@
              <field name="groups_id" eval="[(4, ref('event.group_event_user'))]"/>
      </record>
 
-    <!-- Demo Data for Event Type -->
```
Comment (tde-banana-odoo) [inline addons/event/data/event_demo.xml]: I bet you try to hide a crash when trying to set a column required.

File: addons/mail/views/mail_message_views.xml (hunk 25 above)
Comment (ppa-odoo) [inline addons/mail/views/mail_message_views.xml]: @tde-banana-odoo, 
As per our requirement for digital products, when going to the digital product form to
<code> ir.attachment </code> Kanban view, then after click on the Kanban card, so that attachment of digital product should be downloaded.

File: addons/membership/membership_view.xml (hunk 26 above)
Comment (ppa-odoo) [inline addons/membership/membership_view.xml]: @tde-banana-odoo, Membership details will configure
Membership =>> Configuration=>> Create Member Product, 
Then I removed only fields on view side from product form

File: addons/event/data/event_demo.xml (hunk 29 above)
Comment (ppa-odoo) [inline addons/event/data/event_demo.xml]: @tde-banana-odoo,
I moved some event type demo data in event_sale module, because as per our spec when create event type then create an automatic product of the event type.
And removed <code>event_type_id</code> from <code>product.template</code> and from everywhere removed code related to <code>event_type_id</code>.

File: addons/product/product.py (hunk 28 above)
Comment (ppa-odoo) [inline addons/product/product.py]: @tde-banana-odoo, Committed by @fpodoo

File: addons/product/product.py (hunk 27 above)
Comment (ppa-odoo) [inline addons/product/product.py]: @tde-banana-odoo, Committed by @fpodoo

File: addons/event_sale/models/event.py [hunk 30]
```diff
@@ -6,6 +6,35 @@
...
+    _inherit = 'event.type'
+    _inherits = {"product.product": "product_id"}
+
+    product_id = fields.Many2one('product.product', 'Product', ondelete='cascade', required=True, domain=[('event_ok', '=', True), ('type' '=' 'service')])
```
Comment (tde-banana-odoo) [inline addons/event_sale/models/event.py]: I don't get it ... you creata an event type "Seminar". It creates a product. With a price. That means that all seminars will be based on the same product, whatever the event ?

Hugh.

File: addons/website_sale_digital/product.py [hunk 31]
```diff
@@ -5,4 +5,56 @@
 class product_template(models.Model):
     _inherit = ['product.template']
 
-    digital_content = fields.Boolean('Digital Content', help="If checked, it will allow clients to download the product attachments when they have bought it.")
+    type = fields.Selection([('digital', 'Digital Content'),('product', 'Stockable Product'), ('consu', 'Consumable'), ('service', 'Service')], help="Consumable: Will not imply stock management for this product. \nStockable product: Will imply stock management for this product.\n Digital Product :If select digital , it will allow clients to download the product attachments when they have bought it.")
```
Comment (tde-banana-odoo) [inline addons/website_sale_digital/product.py]: Should have been done with an inherited method. Not hardcoding value, screwing inheritance.

File: addons/stock/procurement.py [hunk 32]
```diff
@@ -111,7 +111,7 @@ def _assign_multi(self, cr, uid, procurements, context=None):
         for procurement in procurements:
             if procurement.rule_id:
                 res[procurement.id] = True
-            elif procurement.product_id.type != 'service':
+            elif procurement.product_id.type not in ['service', 'digital']:
```
Comment (tde-banana-odoo) [inline addons/stock/procurement.py]: That's a nice modular approach ...

File: addons/product/product_view.xml [hunk 33]
```diff
@@ -70,75 +91,69 @@
...
-                                        <field name="weight"/>
-                                        <field name="weight_net"/>
+                                    <group name="lots_and_weight">
+                                       <field digits="(14, 3)" name="weight" string="Weight" invisible="1"/>
                                     </group>
                                 </group>
-                                <group name="packaging" string="Packaging" attrs="{'invisible':[('type','=','service')]}" groups="product.group_stock_packaging" colspan="4">
+                                <separator string="Vendors"/>
+                                <field name="seller_ids"/>
+                                <group name="packaging" string="Packaging" attrs="{'invisible':[('type', 'in', ['service', 'digital'])]}" groups="product.group_stock_packaging" colspan="4">
```
Comment (tde-banana-odoo) [inline addons/product/product_view.xml]: note to myself: attrs inheritance, because inheritance.

File: addons/stock_account/product.py [hunk 34]
```diff
@@ -36,18 +60,23 @@ class product_template(osv.osv):
...
-        'valuation': 'manual_periodic',
-    }
-
-    def onchange_type(self, cr, uid, ids, type):
-        res = super(product_template, self).onchange_type(cr, uid, ids, type)
-        if type in ('consu', 'service'):
-            res = {'value': {'valuation': 'manual_periodic'}}
+    def onchange_type(self, cr, uid, ids, product_type):
+        res = super(product_template, self).onchange_type(cr, uid, ids, product_type)
+        if product_type in ('consu', 'service', 'digital'):
```
Comment (tde-banana-odoo) [inline addons/stock_account/product.py]: note to myself: should not be here

File: addons/stock_account/product.py [hunk 35]
```diff
@@ -129,14 +158,25 @@ def do_change_standard_price(self, cr, uid, ids, new_price, context=None):
 
 class product_product(osv.osv):
     _inherit = 'product.product'
-    def onchange_type(self, cr, uid, ids, type):
-        res = super(product_product, self).onchange_type(cr, uid, ids, type)
-        if type in ('consu', 'service'):
-            res = {'value': {'valuation': 'manual_periodic'}}
+    def onchange_type(self, cr, uid, ids, product_type):
+        res = super(product_product, self).onchange_type(cr, uid, ids, product_type)
+        if product_type in ('consu', 'service', 'digital'):
```
Comment (tde-banana-odoo) [inline addons/stock_account/product.py]: note to myself: should not be here

File: addons/stock_account/product.py [hunk 36]
```diff
@@ -36,18 +60,23 @@ class product_template(osv.osv):
             domain=[('deprecated', '=', False)],
             help="When doing real-time inventory valuation, counterpart journal items for all outgoing stock moves will be posted in this account, unless "
                  "there is a specific valuation account set on the destination location. When not set on the product, the one from the product category is used."),
+        'cost_method': fields.function(_get_cost_method, fnct_inv=_set_cost_method, type='char'),
```
Comment (tde-banana-odoo) [inline addons/stock_account/product.py]: Note: why those fields ?

File: addons/event_sale/models/event.py (hunk 30 above)
Comment (ppa-odoo) [inline addons/event_sale/models/event.py]: @tde-banana-odoo ,
Yes, If create an event type "Seminat" then create an automatic product of event type, that the all seminars will be based on the same product.

File: addons/mrp/mrp_view.xml [hunk 37]
```diff
@@ -562,6 +562,52 @@
                 </search>
             </field>
         </record>
+
+        <record id="product_template_search_view_procurment" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/mrp/mrp_view.xml]: Duplicated views. What's the point ?

File: addons/mrp/mrp_view.xml [hunk 38]
```diff
@@ -562,6 +562,52 @@
...
+             <field name="inherit_id" ref="product.product_template_search_view"/>
+             <field name="arch" type="xml">
+                 <xpath expr="//filter[@name='products']" position="after">
+                     <separator/>
+                    <filter string="Components" name="components" icon="terp-accessories-archiver" domain="[('bom_ids','not in',[])]"/>
+                 </xpath>
+             </field>
+        </record>
+
+        <record id="view_mrp_product_template_form_inherited" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/mrp/mrp_view.xml]: Duplicated views. What's the point ?

File: addons/mrp/mrp_view.xml [hunk 39]
```diff
@@ -562,6 +562,52 @@
...
+                <group name="sale_condition" position="inside">
+                    <label for="produce_delay" attrs="{'invisible':[('type','=','service')]}"/>
+                    <div attrs="{'invisible':[('type','=','service')]}">
+                        <field name="produce_delay" class="oe_inline"/> days
+                    </div>
+                </group>
+            </field>
+        </record>
+        
+        <record id="product_template_action" model="ir.actions.act_window">
```
Comment (tde-banana-odoo) [inline addons/mrp/mrp_view.xml]: Duplicated views. What's the point ?

File: addons/procurement/procurement_view.xml [hunk 40]
```diff
@@ -89,9 +89,9 @@
             <field name="res_model">procurement.order</field>
             <field name="view_type">form</field>
             <field name="view_mode">tree,form</field>
-            <field name="domain">[('group_id','=',active_id)]</field>            
+            <field name="domain">[('group_id','=',active_id)]</field>
```
Comment (tde-banana-odoo) [inline addons/procurement/procurement_view.xml]: This file is full of unnecessary changes.

File: addons/mrp/mrp_view.xml (hunk 37 above)
Comment (ppa-odoo) [inline addons/mrp/mrp_view.xml]: @tde-banana-odoo ,
Yes, This is duplicate view. but, do not need this then I remove it.

File: addons/mrp/mrp_view.xml (hunk 38 above)
Comment (ppa-odoo) [inline addons/mrp/mrp_view.xml]: @tde-banana-odoo ,
Yes, This is duplicate view. but, do not need this then I remove it.

File: addons/mrp/mrp_view.xml (hunk 39 above)
Comment (ppa-odoo) [inline addons/mrp/mrp_view.xml]: @tde-banana-odoo ,
Yes, This is duplicate view. but, do not need this then I remove it.

File: addons/product/product_view.xml [hunk 41]
```diff
@@ -268,26 +254,26 @@
...
-                a physical product, a consumable or a service you offer to
-                customers.
-              </p><p>
-                The product form contains information to simplify the sale
-                process: price, notes in the quotation, accounting data,
-                procurement methods, etc.
-              </p>
-            </field>
-        </record>
+            <field name="help" type="html">
```
Comment (tde-banana-odoo) [inline addons/product/product_view.xml]: Note to myself: unnecessary change

File: addons/product_email_template/models/product.py [hunk 42]
```diff
@@ -14,6 +14,6 @@ class product_template(osv.Model):
         'email_template_id': fields.many2one(
             'mail.template', 'Product Email Template',
             help='When validating an invoice, an email will be sent to the customer'
-                 'based on this template. The customer will receive an email for each'
-                 'product linked to an email template.'),
+                 ' based on this template. The customer will receive an email for each'
```
Comment (tde-banana-odoo) [inline addons/product_email_template/models/product.py]: Note to myself: unnecessary reindentation.

File: addons/product/product_view.xml (hunk 41 above)
Comment (ppa-odoo) [inline addons/product/product_view.xml]: @tde-banana-odoo ,
this changes of indentation.

File: addons/product_email_template/models/product.py (hunk 42 above)
Comment (ppa-odoo) [inline addons/product_email_template/models/product.py]: @tde-banana-odoo ,
This was required for when move cursor on emial_template_id, then show tooltip like :
<code>When validating an invoice, an email will be sent to the ***customerbased*** on this template. 
The customer will receive an email for ***eachproduct*** linked to an email template.</code>

File: addons/stock/product.py [hunk 43]
```diff
@@ -349,29 +346,19 @@ class product_template(osv.osv):
     _inherit = 'product.template'
     
     def _product_available(self, cr, uid, ids, name, arg, context=None):
-        prod_available = {}
```
Comment (tde-banana-odoo) [inline addons/stock/product.py]: Anything really changed ?

Comment (ppa-odoo) [inline addons/stock/product.py]: @tde-banana-odoo , I think remove unnecessary code, committed by @fpodoo

File: addons/stock/wizard/make_procurement_product.py [hunk 44]
```diff
@@ -36,6 +38,30 @@ def onchange_product_id(self, cr, uid, ids, prod_id):
         'qty': lambda *args: 1.0,
     }
 
+    def fields_view_get(self, cr, uid, view_id=None, view_type='form', context=None, toolbar=False, submenu=False):
```
Comment (tde-banana-odoo) [inline addons/stock/wizard/make_procurement_product.py]: Whut ?

Comment (ppa-odoo) [inline addons/stock/wizard/make_procurement_product.py]: @tde-banana-odoo ,
<code>make_procurement</code> model has only <code>product_id</code> and this wizard use for only two view <code>product_template</code> and <code>product_product</code>. 
As per spec if model of view is <code>product.template</code>, then show only template's variants in wizard,
If template has one variant, then product_id field should invisible on wizard,
If model of view is <code>product.product</code> then product_id field should read-only on wizard,

Comment (tde-banana-odoo) [inline addons/stock/wizard/make_procurement_product.py]: It is impossible to understand.

Comment (ppa-odoo) [inline addons/stock/wizard/make_procurement_product.py]: @tde-banana-odoo ,
As per spec
When click on ***Procurement Request*** button, Then 
1: If view is product.template, then should show only variants of current product template's on wizard.
-> If product template has one variant, then product should invisible on wizard,
See here: https://drive.google.com/file/d/0BxqHe5rtIxY0YUl3SEVObmMwXzQ/view

2: If view is product.product then product field should read-only on wizard,

File: addons/delivery/delivery_view.xml [hunk 45]
```diff
@@ -92,6 +92,17 @@
             </field>
         </record>
 
+        <record id="view_template_product_weight" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/delivery/delivery_view.xml]: Note to myself: unnecessary

File: addons/event/data/event_demo.xml (hunk 29 above)
Comment (tde-banana-odoo) [inline addons/event/data/event_demo.xml]: I will revert the whole change in event. It makes no sense to have some kind of default product for completely different event and tickets. Products are linked to tickets, not categories.

File: addons/mail/views/mail_message_views.xml [hunk 46]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0"?>
+<!-- <?xml version="1.0"?> -->
```
Comment (tde-banana-odoo) [inline addons/mail/views/mail_message_views.xml]: Note to myself: typo

File: addons/stock/wizard/make_procurement_product.py (hunk 44 above)
Comment (tde-banana-odoo) [inline addons/stock/wizard/make_procurement_product.py]: And using fields and attributes ? Didn't you think that storing product_tmpl_id and variant_count directly on the wizard could be easier than hacking the fields view get ?

Comment (ppa-odoo) [inline addons/stock/wizard/make_procurement_product.py]: @tde-banana-odoo, you are right,
It is possible to store product_tmpl_id and variant_count directly on the wizard instead of hacking the fields view get.

**Comment by tde-banana-odoo** [comment]
Replaced by https://github.com/odoo/odoo/pull/8124 .
